addons/account/tests/test_account_customer_invoice.py

```diff
@@ -338,3 +338,45 @@ def patched_today(*args, **kwargs):
             dashboard_data = journal.get_journal_dashboard_datas()
             self.assertEquals(dashboard_data['number_late'], 2)
             self.assertIn('78.42', dashboard_data['sum_late'])
+
+    def test_charge_invoice(self):
+        """Create an invoice and then make an extra charge to that invoice"""
+        self.env.user.company_id.tax_calculation_rounding_method = 'round_globally'
+
+        payment_term = self.env.ref('account.account_payment_term_advance')
+        journalrec = self.env['account.journal'].search([('type', '=', 'sale')])[0]
+        partner3 = self.env.ref('base.res_partner_3')
+        account_id = self.env['account.account'].search([('user_type_id', '=', self.env.ref('account.data_account_type_revenue').id)], limit=1).id
+
+        invoice_line_data = [
+            (0, 0,
+                {
+                    'product_id': self.env.ref('product.product_product_1').id,
+                    'quantity': 40.0,
+                    'account_id': account_id,
+                    'name': 'product test 1',
+                    'price_unit': 2.27,
+                }
+             )
+        ]
+
+        invoice = self.env['account.invoice'].create(dict(
+            name="Test Customer Invoice",
+            payment_term_id=payment_term.id,
+            journal_id=journalrec.id,
+            partner_id=partner3.id,
+            invoice_line_ids=invoice_line_data
+        ))
+        invoice.action_invoice_open()
+        wizard = self.env['account.invoice.refund'].with_context(
+            {'active_ids': invoice.ids}).create({
+                'description': 'Charge for unfulfilled payment',
+                'date': datetime.date.today(),
+                'filter_refund': 'charge'
+            })
+        wizard.invoice_refund()
+        charge = self.env['account.invoice'].search([
+            ('origin', '=', invoice.move_name),
+            ('type', '=', 'out_invoice')
+        ])
+        self.assertTrue(charge)
```

Comment:
Add a message if the charge is False, please

Reply:
done